lib/fluent/plugin/buffer/chunk.rb

```diff
@@ -149,11 +152,31 @@ def open(&block)
           raise NotImplementedError, "Implement this method in child class"
         end
 
-        def write_to(io)
+        def write_to(io, options = {})
           open do |i|
             IO.copy_stream(i, io)
           end
         end
+
+        module Decompressable
+          include Fluent::Plugin::Compressable
+
+          def write_to(io, options = {})
+            unless options[:compress] == :gzip
+              c = decompress(read)
+              if @chunk.is_a?(String)
+                @chunk = c
+              else
+                # reset io(@chunk) to overwrite new chunk
+                @chunk.seek(0, IO::SEEK_SET)
+                @chunk.truncate(0)
+                @chunk.write c
```

Comment:
This occurs too heavy I/O operations, and this may cause data loss if `@chunk.write` raises any errors. Don't truncate/overwrite original file chunk.
Can you use StringIO or something others at the same time of decompress operation?